libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -0,0 +1,113 @@
+package com.mapbox.navigation.core.routealternatives
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.navigation.base.route.RouteAlternativesOptions
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.routeoptions.RouteOptionsUpdater
+import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigator
+import com.mapbox.navigation.utils.internal.MapboxTimer
+import com.mapbox.navigation.utils.internal.ThreadController
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.launch
+import java.util.concurrent.CopyOnWriteArraySet
+
+internal class RouteAlternativesController(
+    private val options: RouteAlternativesOptions,
+    private val navigator: MapboxNativeNavigator,
+    private val directionsSession: DirectionsSession,
+    private val tripSession: TripSession,
+    private val routeOptionsUpdater: RouteOptionsUpdater,
+    private val logger: Logger
```

Comment:
No need to take logger as an argument (unless you'd like to test that logs are printed). You can use the `LoggerProvider`.

Reply:
Ah right, I'll update this